app/handlers/signup_start.rb

```diff
@@ -19,30 +19,29 @@ def handle
     if !User.known_roles.include?(signup_params.role)
       fatal_error(code: :unknown_role, offending_inputs: [:signup, :role])
     end
-    outputs.redirect_action = :verify_email
     outputs.role = signup_params.role
 
-    # Return if user went back, didn't change anything, and resubmitted
+    # Return if has not changed email address
     if existing_signup_state.try(:contact_info_value) == email
       existing_signup_state.update_attributes(role: signup_params.role)
       outputs.signup_state = existing_signup_state
-      if trusted_state && trusted_state['email'] == email
-        confirm_trusted_account(existing_signup_state)
-      end
       return
     end
 
     # If email in use, want users to login with that email, not create another account
     fatal_error(code: :email_in_use, offending_inputs: [:signup, :email]) if email_in_use?
 
+    # Create a new one
+    new_signup_state = SignupState.email_address.create(
+      contact_info_value: email,
+      role: signup_params.role,
+      trusted_data: existing_signup_state.trusted_data,
```

Comment:
cool, I like it.

let's not forget too to use the trusted data to improve someone's account (e.g. add verified email if it doesn't exist, help mark faculty verified -- not sure how to do latter yet)

Reply:
Yes, when `signup_state` is initialized from trusted data it's marked as verified, which in turn marks the contact as verified when it's created.

https://github.com/openstax/accounts/pull/539/files/de2966022584438e9a8769bd4bb7dcfa18dc3162..a48de9bfe35eb8672d686157b9f6d0ba7608fa75#diff-6f4447b02666c80e38966472bdc56145R25